client/windows/CMessage.cpp

```diff
@@ -207,6 +227,32 @@ std::vector<std::string> CMessage::breakText( std::string text, size_t maxLineWi
 	return ret;
 }
 
+ColorRGBA CMessage::parseColor(std::string text)
+{
+	std::smatch match;
+	std::regex expr("^#([0-9a-fA-F]{6})$");
+	ui8 rgb[3] = {0, 0, 0};
+	if(std::regex_search(text, match, expr))
+	{
+		std::string tmp = boost::algorithm::unhex(match[1].str()); 
+		std::copy(tmp.begin(), tmp.end(), rgb);
+		return ColorRGBA(rgb[0], rgb[1], rgb[2]);
+	}
+
+	const JsonNode config(JsonPath::builtin("CONFIG/textColors"));
+	auto colors = config["colors"].Struct();
+	for(auto & color : colors) {
+		if(boost::algorithm::to_lower_copy(color.first) == boost::algorithm::to_lower_copy(text))
+		{
+			std::string tmp = boost::algorithm::unhex(color.second.String()); 
+			std::copy(tmp.begin(), tmp.end(), rgb);
+			return ColorRGBA(rgb[0], rgb[1], rgb[2]);
+		}
+	}
+
+	return Colors::TRANSPARENCY;

```

Comment:
Please add warning to log instead of silent return in case of parsing error

Reply:
The reason for silent return is that `{Hello | World}` should still be allowed (and shouldn't be an error/log entry). Only when a color is detected it should be colorized.

Reply:
This would lead to issues if sentence starts with color name.
{Red player has left the game}
Would result in red text 'player has left the game'?

Reply:
Ah, there is also `|` symbol...
Perhaps return std::optional<ColorRGBA> for clarity?

Reply:
Or consider some other formatting form that won't be ambiguous?
e.g.
`{{red}Text colored in red}`